"panic" at runtime since last commit

So since last commit (one hour ago ;) ) i'm facing a "panic" at runtime, not sure yet why ...

```
{"level":"info","ts":1706722906.5507073,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc000827300"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8079be]

goroutine 1 [running]:
go.uber.org/zap.(*Logger).check(0x0, 0x1, {0x214214a, 0x41})
 go.uber.org/zap@v1.26.0/logger.go:330 +0x5e
go.uber.org/zap.(*Logger).Warn(0x7fc9fb0eaff8?, {0x214214a?, 0x6d88e5?}, {0x0, 0x0, 0x0})
 go.uber.org/zap@v1.26.0/logger.go:253 +0x38
github.com/sagikazarmark/caddy-fs-s3.(*FS).Provision(0xc000701740, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000398b80, 0x4, ...}})
 github.com/sagikazarmark/caddy-fs-s3@v0.3.0/module.go:66 +0x55
github.com/caddyserver/caddy/v2.Context.LoadModuleByID({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000398b80, 0x4, 0x4}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:347 +0x458
github.com/caddyserver/caddy/v2.Context.loadModuleInline({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000398b80, 0x3, 0x4}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:397 +0xda
github.com/caddyserver/caddy/v2.Context.LoadModule({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000398b80, 0x3, 0x4}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:162 +0x426
github.com/caddyserver/caddy/v2/modules/caddyhttp/fileserver.(*FileServer).Provision(0xc0006d2a80, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000398b80, 0x3, ...}})
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/fileserver/staticfiles.go:189 +0xcd
github.com/caddyserver/caddy/v2.Context.LoadModuleByID({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000398b80, 0x3, 0x4}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:347 +0x458
github.com/caddyserver/caddy/v2.Context.loadModuleInline({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc0000aedc0, 0x2, 0x2}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:397 +0xda
github.com/caddyserver/caddy/v2.Context.LoadModule({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc0000aedc0, 0x2, 0x2}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:176 +0x6e6
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Route).ProvisionHandlers(0xc0008bc360, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc0000aedc0, 0x2, ...}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/routes.go:154 +0x71
github.com/caddyserver/caddy/v2/modules/caddyhttp.RouteList.ProvisionHandlers({0xc0008bc240, 0x3, 0x1bcac0e?}, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, ...}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/routes.go:209 +0x95
github.com/caddyserver/caddy/v2/modules/caddyhttp.RouteList.Provision({0xc0008bc240, 0x3, 0x4}, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, ...})
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/routes.go:186 +0xb2
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Subroute).Provision(0xc0000aed80, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc0000aedc0, 0x2, ...}})
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/subroute.go:58 +0x5b
github.com/caddyserver/caddy/v2.Context.LoadModuleByID({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc0000aedc0, 0x2, 0x2}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:347 +0x458
github.com/caddyserver/caddy/v2.Context.loadModuleInline({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000864390, 0x1, 0x1}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:397 +0xda
github.com/caddyserver/caddy/v2.Context.LoadModule({{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000864390, 0x1, 0x1}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:176 +0x6e6
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Route).ProvisionHandlers(0xc0008bc120, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000864390, 0x1, ...}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/routes.go:154 +0x71
github.com/caddyserver/caddy/v2/modules/caddyhttp.RouteList.ProvisionHandlers({0xc0008bc000, 0x4, 0x199?}, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, ...}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/routes.go:209 +0x95
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*App).Provision(0xc0006bff40, {{0x28a4b80, 0xc00087ee70}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000864390, 0x1, ...}})
 github.com/caddyserver/caddy/v2@v2.7.6/modules/caddyhttp/app.go:287 +0xc65
github.com/caddyserver/caddy/v2.Context.LoadModuleByID({{0x28a4bb8, 0xc0006eeaf0}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0xc000864390, 0x1, 0x1}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:347 +0x458
github.com/caddyserver/caddy/v2.Context.App({{0x28a4bb8, 0xc0006eeaf0}, 0xc00085e4b0, 0xc0006eea50, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}}, ...)
 github.com/caddyserver/caddy/v2@v2.7.6/context.go:423 +0xee
github.com/caddyserver/caddy/v2.run.func3(...)
 github.com/caddyserver/caddy/v2@v2.7.6/caddy.go:480
github.com/caddyserver/caddy/v2.run(0xc000854098?, 0x1)
 github.com/caddyserver/caddy/v2@v2.7.6/caddy.go:485 +0x4dc
github.com/caddyserver/caddy/v2.unsyncedDecodeAndRun({0xc000856580, 0x508, 0x580}, 0x1)
 github.com/caddyserver/caddy/v2@v2.7.6/caddy.go:338 +0x12a
github.com/caddyserver/caddy/v2.changeConfig({0x20a68bd, 0x4}, {0x20ada9a, 0x7}, {0xc000856000, 0x508, 0x580}, {0x0, 0x0}, 0x1)
 github.com/caddyserver/caddy/v2@v2.7.6/caddy.go:229 +0x773
github.com/caddyserver/caddy/v2.Load({0xc000856000, 0x508, 0x580}, 0x0?)
 github.com/caddyserver/caddy/v2@v2.7.6/caddy.go:128 +0x23c
github.com/caddyserver/caddy/v2/cmd.cmdRun({0x0?})
 github.com/caddyserver/caddy/v2@v2.7.6/cmd/commandfuncs.go:231 +0x838
github.com/caddyserver/caddy/v2/cmd.init.1.func2.WrapCommandFuncForCobra.func1(0xc00078a300, {0x20a6921?, 0x4?, 0x20a68ed?})
 github.com/caddyserver/caddy/v2@v2.7.6/cmd/cobra.go:137 +0x2f
github.com/spf13/cobra.(*Command).execute(0xc00078a300, {0xc0002b5660, 0x2, 0x2})
 github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0x3891c20)
 github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
 github.com/spf13/cobra@v1.7.0/command.go:992
github.com/caddyserver/caddy/v2/cmd.Main()
 github.com/caddyserver/caddy/v2@v2.7.6/cmd/main.go:66 +0x5b
main.main()
 caddy/main.go:15 +0xf
```

With a configuration that was working (1 hour ago) and being rather simple : 


```
route /logs/* {
	uri strip_prefix /logs

	route @gz {
		header Content-Type application/json
		header Content-Encoding gzip
	}

	file_server browse {
		fs s3 {
			bucket "XXXXXXXXXXXXXXXXXXXXX"
			region "YYYYYYYYYYYYYYYYYYYYYYY"
			force_path_style
		}
		pass_thru
	}
}
```

--- a/module.go
+++ b/module.go
@@ -45,8 +45,6 @@ type FS struct {
 	// DEPRECATED: please use 'use_path_style' instead.
 	// Set this to `true` to force the request to use path-style addressing.
 	S3ForcePathStyle bool `json:"force_path_style,omitempty"`
-
-	logger *zap.Logger
 }
 
 // CaddyModule returns the Caddy module information.
@@ -63,7 +61,7 @@ func (fs *FS) Provision(ctx caddy.Context) error {
 	}
 
 	if fs.S3ForcePathStyle {
-		fs.logger.Warn("force_path_style is deprecated, please use use_path_style instead")
+		ctx.Logger().Warn("force_path_style is deprecated, please use use_path_style instead")
 	}
 
 	var configOpts []func(*config.LoadOptions) error
@@ -78,7 +76,8 @@ func (fs *FS) Provision(ctx caddy.Context) error {
 
 	cfg, err := config.LoadDefaultConfig(ctx.Context, configOpts...)
 	if err != nil {
-		fs.logger.Error("could not create AWS config", zap.Error(err))
+		ctx.Logger().Error("could not create AWS config", zap.Error(err))
+
 		return err
 	}
 

